tests/repeated-updates.sh

```diff
@@ -0,0 +1,39 @@
+#! /bin/sh -e
+
+SCRATCH=scratch/$(basename $0 .sh)
+PATCHELF=$(readlink -f "../src/patchelf")
+
+rm -rf ${SCRATCH}
+mkdir -p ${SCRATCH}
+
+cp simple ${SCRATCH}/
+cp libfoo.so ${SCRATCH}/
+cp libbar.so ${SCRATCH}/
+
+cd ${SCRATCH}
+
+${PATCHELF} --add-needed ./libbar.so simple
+
+###############################################################################
+# Test that repeatedly modifying a string inside a shared library does not
+# corrupt it due to the addition of multiple PT_LOAD entries
+###############################################################################
+load_segments_before=$(readelf -W -l libbar.so | grep -c LOAD)
+
+for i in $(seq 1 100)
+do
+    ${PATCHELF} --set-soname ./libbar.so libbar.so
+    ${PATCHELF} --set-soname libbar.so libbar.so
+    ./simple || exit 1
+done
+
+load_segments_after=$(readelf -W -l libbar.so | grep -c LOAD)
+
+###############################################################################
+# To be even more strict, check that we don't add too many extra LOAD entries
+###############################################################################
+echo "Segments before: ${load_segments_before} and after: ${load_segments_after}"
+if ((${load_segments_after} > ${load_segments_before} + 2 ))
```

Comment:
For our openbsd and freebsd friends:

```suggestion
if [ ${load_segments_after} -gt ${load_segments_before} + 2 ]
```

Reply:
We should have a rule on "make check" that runs shellcheck :D

Reply:
I would but than testing shellcheck (ghc) on some more exotic platforms (mips64, sparc64...) is not possible. I think it's good enough if it's part of the ci.

Reply:
Yes, makes more sense.
Maybe we could have a `make premerge-check` which can be used by the CI and developers